### ISSUE TITLE
SDK3?

### ISSUE THREAD

**Issue by Phr3d13** [SCRN-VRC/Chess-Shader-With-Artificial-Intelligence#2]
Is there anything stopping this from working in a Udon/SDK3 world?

**Comment by SCRN-VRC** [comment]
No, you can strip off all the SDK2 stuff, that's only used to sync a ball that you grab. Everything else should be portable since its just shaders.

**Comment by Phr3d13** [comment]
any chance you'd feel like updating it? I'm a complete Unity noob...

**Comment by SCRN-VRC** [comment]
You can try using the avatar's prefab for a world, that has got sdk2 stuff removed cause avatars can't them.

**Comment by Phr3d13** [comment]
Ok, that one seems like it works better... but I'm either having issues understanding how to play, or the controls are having issues. I can walk through the board and get boxes to light up green and orange, but can't figure out how to actually start playing.

**Comment by SCRN-VRC** [comment]
Ah it's VR only, you have to tap the board on what you want to move then tap where you want to move to. That's why there's a ball in the world version, to simulate a hand touching the board.

**Comment by Phr3d13** [comment]
and it's the ball in the world version that is locked to SDK2?

**Comment by SCRN-VRC** [comment]
You can try adding a synced object in Udon that's people can grab

**Comment by Phr3d13** [comment]
Exactly what I needed for K/M (non-VR) interacting! Thank you!

**Comment by Phr3d13** [comment]
I used a Marker (from the free pool/billiards game unity asset that I was already using), added 4 components to it (Rigidbody (added when adding one of the other components (turned off "Use Gravity", and turned on "Is Kinematic")), Mesh Collider (had to turn on Convex), Udon Behavior (turned on "Synchronize Position" and "Allow Ownership Transfer on Collision"), and VRC Pickup).

**Comment by SCRN-VRC** [comment]
I can try making it more desktop friendly by just letting people click on the board. Though this probably won't happen any time soon.
